src/Faker/Core/Barcode.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Core;
+
+use Faker\Extension;
+
+/**
+ * @experimental This class is experimental and does not fall under our BC promise
+ */
+final class Barcode implements Extension\BarcodeExtension
+{
+    /**
+     * @param int $length
+     */
+    private function ean($length = 13): string
+    {
+        $code = Extension\Helper::numerify(str_repeat('#', $length - 1));
+
+        return $code . \Faker\Calculator\Ean::checksum($code);
```

Comment:
I maybe would have done this with sprintf to keep it more readable and a bit more secure overall.

Reply:
I've changed this now.